internal/pkg/manifest/workload_ecs.go

```diff
@@ -85,13 +85,57 @@ type ImageWithPortAndHealthcheck struct {
 	HealthCheck   ContainerHealthCheck `yaml:"healthcheck"`
 }
 
+// AlarmArgs represents specs of CloudWatch alarms for deployment rollbacks.
+type AlarmArgs struct {
+	CPUUtilization    *int `yaml:"cpu_utilization"`
+	MemoryUtilization *int `yaml:"memory_utilization"`
+}
+
+// RollbackAlarmArgsOrNames represents alarms for rolling back deployments.
+// It supports unmarshalling yaml which can either be of type AlarmArgs or a list of strings.
+type RollbackAlarmArgsOrNames struct {
+	AlarmNames []*string
+	AlarmArgs
+}
+
+// UnmarshalYAML overrides the default YAML unmarshaling logic for the RollbackAlarmArgsOrNames
+// struct, allowing it to perform more complex unmarshaling behavior.
+// This method implements the yaml.Unmarshaler (v3) interface.
+func (r *RollbackAlarmArgsOrNames) UnmarshalYAML(value *yaml.Node) error {
```

Comment:
Can we try using Danny's Union implementation here? That will help get rid of this method.

Reply:
done